### ISSUE TITLE
File upload: Component override onClick prop

### ISSUE THREAD

**Issue by metalice** [patternfly/patternfly-react#5417]
Fileupload component is overring the prop of onClick.

This event is very helpful and cannot be used because of the override.

in file [Fileupload.tsx](https://github.com/patternfly/patternfly-react/blob/master/packages/react-core/src/components/FileUpload/FileUpload.tsx) around line 131 is the override:
```javascript
<FileUploadField
          {...getRootProps({
            ...props,
            refKey: 'containerRef',
            onClick: event => event.preventDefault() // Prevents clicking TextArea from opening file dialog
          })}
          tabIndex={null}
.....
```

I'm suggesting to change it to something like that:

```javascript
onClick: event => { 
    event.preventDefault();
    props?.onClick(event); 
}
```

**Comment by kmcfaul** [comment]
Upon testing, it looks like the onClick passed to the FileUploadField actually does nothing because it gets set on the div (and wasn't triggering). I adjusted the onClick to pass to FileUploadField's text area directly as a callback, and added a property to FileUploadField as well. @metalice Is this the behavior of the onClick you were looking for?

**Comment by nicolethoen** [comment]
This PR could possibly be exposing a few more handlers to help this situation? 
https://github.com/patternfly/patternfly-react/pull/6139

**Comment by nicolethoen** [comment]
Addressed by [PR #5689](https://github.com/patternfly/patternfly-react/pull/5689)
